modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpResponse.java

```diff
@@ -45,8 +47,8 @@ public boolean containsHeader(String name) {
         return headers().contains(name);
     }
 
-    public Netty4HttpRequest getRequest() {
-        return request;
+    public HttpHeaders requestHeaders() {
+        return requestHeaders;
     }
```

Comment:
sorry maybe a naive comment, is it better to name public api `getRequestHeaders` ?